src/rulesets/Random/random.jl

```diff
@@ -6,3 +6,16 @@ function rrule(::Type{MersenneTwister}, args...)
     end
     return MersenneTwister(args...), MersenneTwister_pullback
 end
+
+@non_differentiable Broadcast.broadcastable(::AbstractRNG)
+
+@non_differentiable Random.randexp(::AbstractRNG)
+@non_differentiable Random.randstring(::AbstractRNG)
+
```

Comment:
```suggestion

@non_differentiable rand()
```

Reply:
This has no inputs.
Isn't it implict that it is not differentiable?

Reply:
Kind of. An AD system would have to figure out that `rand` has no internal data and deduce that it can ignore it. This way if the AD system isn't very good at doing that things are still okay.

Reply:
I think I would like to cover all function without any args in a seperate PR.

Reply:
But I guess we can do this one since you've written it now and bad things happen when you bring your AD too close to random numbers